test/nextcloud_test.dart

```diff
@@ -114,27 +122,173 @@ void main() {
     });
+    test('Move file (no overwrite)', () async {
+      final path = '${config.testDir}/move-test.txt';
+      final data = utf8.encode('WebDAV movetest');
+      await client.webDav.upload(data, path);
+
+      expect(
+          () => client.webDav.move(
+              '${config.testDir}/test.txt', '${config.testDir}/move-test.txt',
+              overwrite: false),
+          throwsA(predicate((e) => e.statusCode == 412)));
+    });
+    test('Move file (overwrite)', () async {
+      final path = '${config.testDir}/move-test.txt';
+      final data = utf8.encode('WebDAV movetest');
+      await client.webDav.upload(data, path);
+
+      final response = await client.webDav.move(
+          '${config.testDir}/test.txt', '${config.testDir}/move-test.txt',
+          overwrite: true);
+      expect(response.statusCode, 204);
+    });
+    test('Get file properties', () async {
+      final startTime = DateTime.now().subtract(Duration(seconds: 2));
+      final path = '${config.testDir}/prop-test.txt';
+      final data = utf8.encode('WebDAV proptest');
+      await client.webDav.upload(data, path);
+
+      final file = await client.webDav.getProps(path);
+      expect(file.isDirectory, false);
+      expect(file.name, 'prop-test.txt');
+      expect(file.lastModified.isAfter(startTime), isTrue,
+          reason: 'Expected lastModified: $startTime < ${file.lastModified}');
+      expect(file.uploadedDate.isAfter(startTime), isTrue,
+          reason: 'Expected uploadedDate: $startTime < ${file.uploadedDate}');
+      expect(file.mimeType, 'text/plain');
+      expect(file.path, path);
+      expect(file.shareTypes, isEmpty);
+      expect(file.size, data.length);
+    });
+    test('Get directory properties', () async {
+      final path = Uri.parse(config.testDir);
+      final file = await client.webDav.getProps(path.toString());
+      expect(file.isDirectory, true);
+      expect(file.isCollection, true);
+      expect(file.name, path.pathSegments.last);
+      expect(file.lastModified, isNotNull);
+      expect(file.mimeType, isNull);
+      expect(file.path, '$path/');
+      expect(file.shareTypes, isEmpty);
+      expect(file.size, greaterThan(0));
+    });
+    test('Get additional properties', () async {
+      final path = '${config.testDir}/prop-test.txt';
+      final file = await client.webDav.getProps(path);
+
+      expect(file.getOtherProp('oc:comments-count', 'http://owncloud.org/ns'),
+          '0');
+      expect(file.getOtherProp('nc:has-preview', 'http://nextcloud.org/ns'),
+          'true');
+    });
+    test('Filter files', () async {
+      final path = '${config.testDir}/filter-test.txt';
+      final data = utf8.encode('WebDAV filtertest');
+      final response = await client.webDav.upload(data, path);
+      final id = response.headers['oc-fileid'];
+
+      // Favorite file
+      await client.webDav.updateProps(path, {'oc:favorite': '1'});
```

Comment:
This should use the constant

Reply:
Good point, updated. I left a few non-constant uses in place in the tests to emphasise that it's just a string.